### PR TITLE
GG-30750: Statistics views

### PR DESCRIPTION
**Pull request by Berkof**
For now, it targeted to 18641, but in general, after 18641's merge it will be remapped to master.

### CODE REVIEW

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/ColumnStatisticsCollector.java [hunk 1]
```diff
@@ -259,7 +260,8 @@ public static ColumnStatistics aggregate(Comparator<Value> comp, List<ColumnStat
         int averageSize = averageSize(totalSize, total, nullsCnt);
 
         return new ColumnStatistics(min, max, nullsPercent(nullsCnt, total),
-                cardinalityPercent(nullsCnt, total, hll.cardinality()), total, averageSize, hll.toBytes(), ver);
+            cardinalityPercent(nullsCnt, total, hll.cardinality()), total, averageSize, hll.toBytes(), ver,
+            U.currentTimeMillis());
```
Comment (tledkov-gridgain) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/ColumnStatisticsCollector.java:264]: I don't quite agree. What do you think about `max(partStat.createdAt())`?

Comment (Berkof) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/stat/ColumnStatisticsCollector.java:264]: agree

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsViewsTest.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+/**
+ * Tests for statistics related views.
+ */
+public abstract class StatisticsViewsTest extends StatisticsAbstractTest {
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+        cleanPersistenceDir();
+
+        //startGridsMultiThreaded(1);
```
Comment (tledkov-gridgain) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsViewsTest.java:34]: debug?

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsViewsTest.java [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+     * @param actual Actual lines.
+     */
+    private void checkContains(List<List<Object>> expected, List<List<?>> actual) {
+        assertTrue(expected.size() <= actual.size());
+
+        assertTrue("Test may take too long with such datasets of actual = " + actual.size(), actual.size() <= 1024);
+
+        for (List<Object> exp : expected) {
+            boolean found = false;
+            for (List<?> act : actual) {
```
Comment (tledkov-gridgain) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsViewsTest.java:86]: Empty lines at this block

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsViewsTest.java (hunk 2 above)
Comment (Berkof) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsViewsTest.java:34]: yep

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsViewsTest.java (hunk 3 above)
Comment (Berkof) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/stat/StatisticsViewsTest.java:86]: done

**Pull request merged**